### PR TITLE
[release-1.11] feat: annotations on gateway serviceaccount

### PR DESCRIPTION
**Pull request by bryankaraffa**
**Please provide a description of this PR:**
Manual cherrypick required.

Resolves the issue `#33914 failed to apply on top of branch "release-1.11"` on https://github.com/istio/istio/issues/34838

### CODE REVIEW

**Review by jwendell — COMMENTED** [review]
@bryankaraffa it looks like this is a new feature? We generally don't backport new features to stable releases.

**Comment by bryankaraffa** [comment]
Hi @jwendell -- understood.  The new feature was opted for cherry-picking by @jacob-delgado for both [release-1.10](https://github.com/istio/istio/pull/34969) and [1.11](https://github.com/istio/istio/pull/34969)

Please  let me know if there is anything else I can help to get this approved and merged still?  Thanks in advance

**Comment by jacob-delgado** [comment]
I wouldn't consider this a new feature as the charts should have been included when they were first created. It does block some people from using easily adding annotations on gateways; users not being able to adapt gateways per their environment. However, I'll defer to the 1.11 RMs.

File: releasenotes/notes/34962.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: release-notes/v2
+kind: feature
+area: security
+
+# issue is a list of GitHub issues resolved in this note.
+issue:
+- 34838
```
Comment (jwendell) [inline releasenotes/notes/34962.yaml:7]: This doesn't look right. This should point to an actual issue, not to the failed cherry-pick. If there's none, leave it blank, although, as @jacob-delgado suggested if this PR fixes a bug, we should have a bug filed.

File: releasenotes/notes/34962.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+docs: []
+
+releaseNotes:
+- |
+  **Added** values to the Istio Gateway Helm charts for configuring annotations on the ServiceAccount.  Can be used to enable [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) on AWS EKS.
```
Comment (jwendell) [inline releasenotes/notes/34962.yaml:13]: If this is a bug fix, it should be **Fixed**

**Comment by bryankaraffa** [comment]
/ok-to-test

Marking OK to test for now to bring parity with PR for `release-1.10` [ PR #34969 ]..  It's been approved for 1.10 it seems, but I don't think it makes sense to merge release-1.10 PR unless also approved merge into release-1.11 as well.  We are currently on 1.10, plan to upgrade to 1.11 soon after some testing

Please let me know if there are any still concerns.  Here's the original PR #33914 explaining the use-case around needing annotations on the gateway serviceaccount.

**Comment by bryankaraffa** [comment]
@jwendell -- It looks like the [PR for release-1.10](https://github.com/istio/istio/pull/34969) was auto merged..  Hoping this is a change that can still be included in release-1.11 as well.  Please let me know if you have any requested changes to the release notes or anything here.  

Thanks in advance all!

**Comment by jwendell** [comment]
@bryankaraffa See my comment on the release note bug number item.

**Comment by bryankaraffa** [comment]
> @bryankaraffa See my comment on the release note bug number item.

Remove issue number in eec8096 so release notes match what is in [`master`](https://github.com/istio/istio/blob/master/releasenotes/notes/helm_chart_gateway_serviceaccount_annotations.yaml)

**Review by jwendell — APPROVED** [review]

**Pull request merged**